Spigot-Server-Patches/0348-Cached-IBlockData-to-CraftBlockData-conversion.patch

```diff
@@ -0,0 +1,76 @@
+From 288b2d33b5f0d069d52f08415a7389823f3c088e Mon Sep 17 00:00:00 2001
+From: miclebrick <miclebrick@outlook.com>
+Date: Thu, 23 Aug 2018 11:45:32 -0400
+Subject: [PATCH] Cached IBlockData to CraftBlockData conversion
+
+
+diff --git a/src/main/java/net/minecraft/server/BlockData.java b/src/main/java/net/minecraft/server/BlockData.java
+index 4dd6c3276..3449be6fe 100644
+--- a/src/main/java/net/minecraft/server/BlockData.java
++++ b/src/main/java/net/minecraft/server/BlockData.java
+@@ -1,6 +1,7 @@
+ package net.minecraft.server;
+ 
+ import com.google.common.collect.ImmutableMap;
++import org.bukkit.craftbukkit.block.data.CraftBlockData;
+ 
+ public class BlockData extends BlockDataAbstract<Block, IBlockData> implements IBlockData {
+ 
+@@ -11,4 +12,14 @@ public class BlockData extends BlockDataAbstract<Block, IBlockData> implements I
+     public Block getBlock() {
+         return (Block) this.e_;
+     }
++
++    // Paper start - impl cached craft block data, lazy load to fix issue with loading at the wrong time
++    private CraftBlockData cachedCraftBlockData;
++
++    @Override
++    public CraftBlockData getCachedCraftBlockData() {
++        if(cachedCraftBlockData == null) cachedCraftBlockData = java.util.Objects.requireNonNull(CraftBlockData.createData(this));
```

Comment:
When would `createData` return null?

Reply:
Dunno, but I had some weird issues and if it happens again this way it'll show up earlier and be easier to debug

Reply:
Well, as far as I can tell it is impossible that the method returns null, so this seems redundant.

Reply:
fine I'll remove it :P